### ISSUE TITLE
Sometimes camera view gets wobbly and blurry

### ISSUE THREAD

**Issue by iNima** [AgoraIO/Flutter-SDK#196]
**Describe the bug**
Sometimes the video gets wobbly and blurry. It only gets back to normal when we switch the camera or restart the app.
Please check the attached videos.

**To Reproduce**
Unfortunately, this issue is intermittent but usually, it happens after a couple of minutes of streaming.

**Expected behavior**
Show the camera view without blurriness or wavy effect.

**Smartphone (please complete the following information):**
 - Device: iPhone Xs
 - OS: iOS 14.0.1
 - Agora SDK Version 3.1.2
- Flutter Version: 1.22.0

```
    VideoEncoderConfiguration config = VideoEncoderConfiguration();
    config.dimensions = VideoDimensions(videoWidth, videoHeight);
    config.orientationMode = VideoOutputOrientationMode.Adaptative;
    config.frameRate = VideoFrameRate.Fps30;
    config.bitrate = videoBitRate;
```

**Captured videos**
[video1.zip](https://github.com/AgoraIO/Flutter-SDK/files/5352511/video1.zip)
[video2.zip](https://github.com/AgoraIO/Flutter-SDK/files/5352515/video2.zip)

**Comment by plutoless** [comment]
@iNima it would be nice if you can also provide the channel name you tested and the timestamp when the issue happened, so that we can check out from our backend.

**Comment by iNima** [comment]
@plutoless The channel name is set dynamically via our backend server every time we start streaming so we can't grab it easily when the issue happens. 
Do you think sending the AgoraSdkLog file could help you in finding the issue?

**Comment by iNima** [comment]
@plutoless I could get the agorasdklog file. Based on the content of the file the channel id is: `bbf13a1f-c028-45b9-a32f-5eb74c800cc5` uid is : `1834556646` 
[agorasdk.log.zip](https://github.com/AgoraIO/Flutter-SDK/files/5362362/agorasdk.log.zip)

**Comment by plutoless** [comment]
@iNima 
![image](https://user-images.githubusercontent.com/471561/96077637-f0c87980-0ee2-11eb-9821-25941cd3ebb4.png)
![image](https://user-images.githubusercontent.com/471561/96077745-31c08e00-0ee3-11eb-91f4-feb4606abd1a.png)

it seems your iOS is constantly sending 1280x720 video. While your Web client was having some network quality issues and the resolution auto adjusted to adapt the network bandwidth estimation. from the image you can see the resolution dropped to 960x540, so this is why you felt the video got wobbly and blurry.

**Comment by iNima** [comment]
What I see on iOS is the local video, though. 
You mean the local video resolution changes automatically to adapt to the network bandwidth?
If this is the case what would be the solution? 

Thanks.

**Comment by plutoless** [comment]
you are saying the iOS local video gets wobbly and blurry? do you have videos recordings of that? that's theoretically impossible.

**Comment by iNima** [comment]
@plutoless Yes, two videos that I attached to my first post above have been captured from the iPhone device. Please check it out.

**Comment by plutoless** [comment]
@iNima did you use any filters? i watched the video and it really looks weird (the image floats like water..) if you are talking about this, it more looks like a filter effect or camera zoom issue, this is the only possibility if you are talking about local video blurry.

**Comment by iNima** [comment]
@plutoless Yes, exactly, the issue is weird and it is like a special effect although I didn't add any filter or effect to it!

**Comment by plutoless** [comment]
is there anything wrong with system camera?
also can you try https://github.com/AgoraIO/Basic-Video-Call/tree/master/One-to-One-Video/Agora-iOS-Tutorial-Swift-1to1 this demo to see if it's working properly?

**Comment by iNima** [comment]
@plutoless  A similar issue happened again, this time on another iPhone.
Please see the attached video.
Just please keep in mind that the video that I attached is not the received streaming video, it is the local video of the client who is streaming to others.
[Image from iOS.MOV.zip](https://github.com/AgoraIO/Flutter-SDK/files/5522721/Image.from.iOS.MOV.zip)

**Comment by littleGnAl** [comment]
If you still face issues, please try upgrading to the new version to see if it works for you.
